android/src/main/io/grpc/android/AndroidChannelBuilder.java

```diff
@@ -0,0 +1,108 @@
+package io.grpc.android;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+import io.grpc.ChannelBuilder;
+import io.grpc.ChannelImpl;
+import io.grpc.ClientInterceptor;
+import io.grpc.ExperimentalApi;
+import io.grpc.okhttp.NegotiationType;
+import io.grpc.okhttp.OkHttpChannelBuilder;
+
+/**
+ * A {@link io.grpc.ChannelBuilder} that provides a stable interface for producing
+ * {@link io.grpc.Channel} instances on Android. This allows applications to be resilient
+ * to changes such as defaulting to a new channel implementation or breaking API changes
+ * in existing ones. Configuration options will be added here only if they provide value
+ * to a broad set of users and are likely to be resilient to changes in the underlying channel
+ * implementation.
+ *
+ * <p>Developers needing advanced configuration are free to use the underlying channel
+ * implementations directly while assuming the risks associated with using an
+ * {@link io.grpc.ExperimentalApi}.
+ */
+
+// TODO(lryan):
+// - Document the various dangers of scheduling work onto Android Main/UI/whatever
+// threads. Is any enforcement practical? Conversely is there a sensible default. I assume
+// its also a bad idea to use huge numbers of threads too.
+// - Provide SSL installation and detect ALPN/NPN support?
+// - Allow for an alternate TrustManager ? for self-signed, pinning etc. Is there a good default
+//   impl for these variations blessed by security team?
+// - Augment with user-agent with app id?
+// - Specify a smaller flow-control window by default? Fewer concurrent streams?
+public class AndroidChannelBuilder extends ChannelBuilder<AndroidChannelBuilder> {
+
+  // Currently we only support OkHttp on Android. This will become dynamic as more transport
+  // implementations become available.
+  private final OkHttpChannelBuilder baseBuilder;
```

Comment:
Should we expose it? If a user wants to have a "better to have" advanced configuration, he may want to do something like:
```
if (androidBuilder.getBaseBuilder() instanceof OkHttpChannelBuilder) {
  ((OkHttpChannelBuilder) androidBuilder.getBaseBuilder()).connectionSpec();
}
```